Services/Catalogue/Mappers/ConfigurableProductsData.php

```diff
@@ -266,6 +277,32 @@ private function getCategoryArray($category)
             $parentUrlKey = '';
         }
 
+        if ($this->allStoreBaseUrls === null) {
+            $this->allStoreBaseUrls = $this->storeManager->getAllStoreBaseUrls();
+        }
+
+        if (!isset($this->currentStoreBaseUrl[$this->storeId])) {
+            $this->currentStoreBaseUrl[$this->storeId] = $this->storeManager->getCurrentStoreBaseUrl();
+            ;
+        }
+
+        if ($this->currentStoreBaseUrl[$this->storeId] &&
+            isset($this->currentStoreBaseUrl[$this->storeId]['base_url'])
+        ) {
+            $currentStoreBaseUrl = $this->currentStoreBaseUrl[$this->storeId];
+            $currentStoreBaseUrl = $currentStoreBaseUrl['base_url'];
+        }
+
+        $categoryUrl = $category->getUrl();
+        foreach ($this->allStoreBaseUrls as $store) {
+            if (strpos($categoryUrl, $store['base_url']) === 0) {
+                $categoryUrl = str_replace($store['base_url'], '', $categoryUrl);
+                break;
+            }
+        }
+
+        $categoryUrl = $currentStoreBaseUrl."".$categoryUrl;
```

Comment:
What if we did not find any URL and replaced it? Only append $currentStoreBaseUrl in the if condition above. And only if $currentStoreBaseUrl is set or not null